build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,169 @@
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
+
+echo "Uploading the packages.."
+if [ "$os" == "ubuntu" ]; then
+        cd ..
+        dput ppa:gluster/glusterfs-${series} glusterfs_${version}-${os}1~${flavor}1_source.changes
+
+    echo "Done"
+    exit
+fi
+
+# we are using the same builder machine to build so we are running the "pbuilder
+# create" everytime to create the chroot according to the os and flavor we want to build.
+echo "creating chroot for ${os} ${flavor}"
+sudo pbuilder create --distribution ${flavor} --mirror ${mirror} --debootstrapopts --keyring=/usr/share/keyrings/${os}-archive-keyring.gpg
+
+echo "Building glusterfs-${version} for ${os} ${flavor} using the chroot and .dsc we created"
+
+# have to use the .dsc file inside the ${os}${flavor} folder
+sudo pbuilder build ~/${os}-${flavor}-Glusterfs-${version}/build/glusterfs_${version}-${release}.dsc | tee build.log
+
+#move the packages to packages directory.
+cp /var/cache/pbuilder/result/glusterfs*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
+rm -rf /var/cache/pbuilder/result/glusterfs*${version}-${release}*.deb
+
+if [ "$flavor" != "stretch" ]; then
+     mv /var/cache/pbuilder/result/libg*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
+fi
+/usr/share/debdelta/dpkg-sig -v -k ${pbuild_key} --sign builder ~/${os}-${flavor}-Glusterfs-${version}/packages/glusterfs-*${version}-${release}*.deb
+
+cd /var/www/repos/apt/debian/
+
+rm -rf pool/* dists/* db/*
+
+cp ~/conf.distributions/${series} conf/distributions
+
+# distribute Debian packages using apt
+for i in ~/${os}-${flavor}-Glusterfs-${version}/packages/glusterfs-*${version}-${release}*; do reprepro includedeb $flavor $i; done
+if [ "$flavor" != "stretch" ]; then
+     for i in ~/${os}-${flavor}-Glusterfs-${version}/packages/libg*${version}-${release}*.deb; do reprepro includedeb $flavor $i; done
+fi
+reprepro includedsc ${flavor} ~/${os}-${flavor}-Glusterfs-${version}/build/glusterfs_${version}-${release}.dsc
+
+tar czf ~/${os}-${flavor}-Glusterfs-${version}/${flavor}-apt-amd64-${version}.tgz pool/ dists/
+
+echo "Pushing Changelog changes.."
+git push origin ${flavor}-${series}-local:${flavor}-glusterfs-${series}
+
+cd ~/${os}-${flavor}-Glusterfs-${version}
+
+#copy the tar.gz file produced by the build to download.rht.gluster.org:/var/www/scratch
+scp $flavor-apt-amd64-$version.tgz gluster_ant@download.rht.gluster.org:/var/www/scratch
+
+ssh gluster_ant@download.rht.gluster.org /var/www/html/pub/gluster/unpacking-script.sh series version os flavor latest_version latest_series
+
+cd ..
+function finish {
+  # cleanup code
+  echo "removing the chroot"
+  sudo rm -rf /var/cache/pbuilder/base.tgz
+
+  #removing folders created while packaging
+  rm -rf ~/${os}-${flavor}-Glusterfs-${version}
+}
+trap finish EXIT
```

Comment:
Traps should be placed at the start of the script. Otherwise, it be used only when the script is over, which is not useful.

Reply:
fixed